### ISSUE TITLE
Investigate harnessing dependabot to automate updates

### ISSUE THREAD

**Issue by numist** [numist/highlight-css#6]
Currently, I watch Pygments and Rouge and manually update the repo whenever either releases a new version. Which is fine. But if I fix #5 then it really just comes down to running `make; git commit -am` on `main` and `gh-pages`—is there some way I can get @dependabot to kick off an action that does this?

**Comment by indirect** [comment]
Yes, create a workflow that automatically merges Dependabot PRs to `main`. Then, create a GitHub action that runs on every commit to main that runs `make` and commits to `gh-pages`. Should do it, I think?

**Comment by numist** [comment]
So schematically I'd have something like:

1. Convince @dependabot that this repo depends on both Pygments (a Python dependency) *and* Rouge (Ruby)
1. New GitHub Action on `pull_request` events (default activity types) targeting `main` by @dependabot calls `make; git commit -am; git push` and merges the PR if nothing broke
1. Another GitHub Action on `commit` to `main` kicks off a `make; git commit -am; git push` to `gh-pages` (assuming nothing breaks)

**Comment by indirect** [comment]
For Rouge, you can do that with `bundle init; bundle add rouge; git add Gemfile*; git commit -m "add rouge to the gemfile"`.  Then one dependabot config section per package manager. I don't know the idiomatic way to do it for python though, sorry.

**Comment by numist** [comment]
[Looks like](https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/about-the-dependency-graph#supported-package-ecosystems) Python (well, `pip`) dependencies are supported via `requirements.txt` and/or `pipfile.lock`

**Comment by numist** [comment]
It occurs to me that I don't actually need two branches and everything currently in `main` can be rolled into `gh-pages`.

**Comment by indirect** [comment]
In that case you can also just tell GitHub to actively publish the main branch as a site!
